### DISCUSSION TITLE
About damaged image mended

### DISCUSSION THREAD

**Discussion by jishi92** [libvips/libvips#3260] [Q&A]
If I load a damaged image and convert its format. The damaged part will be mended by copy and I guess is VIPS_EXTEND_COPY.
But I can't find where to change it ,such as VIPS_EXTEND_BLACK.

The case:
![broken-load](https://user-images.githubusercontent.com/22196414/210359731-39b4872d-abe0-4bc9-9993-c04127296594.jpeg)

I use :
vips copy broken-load.jpeg xxxtest.webp

**Comment by jishi92** [comment]
the webp image can't upload

**Comment by jcupitt** [comment]
That's a PNG image, so the missing part is just transparent. You could fix it with:

```
vips flatten ring.png x.jpg --background 255
```

Truncated JPEG images are more difficult since they don't have transparency. It would need a change to libvips to fix this, perhaps a `background` option to `jpegload`.

**Reply by jishi92** [reply]
Ok,thanks for your reply.
Look forward to libvips early implementation of this function.
